activities/activity_arielsaffer.py

```diff
@@ -0,0 +1,83 @@
+import grass.script as gs
+import datetime
+
+
+# compute solar irradiance (W/m2) for a given day and hour
+# and extract the shades cast by topography, using the time zone
+# of the surface location (based on NE corner), if packages are
+# available (otherwise, current timezone is used)
+def run_radiance_now(scanned_elev, env, **kwargs):
+    try:
+        # you may need to install this package
+        import pytz
+
+        # you probably need to install this package
+        from timezonefinder import TimezoneFinder
+
+        obj = TimezoneFinder()
+
+        # Get the current day/time
+        today = datetime.datetime.today()
+        # Get the lat and lon of the NE corner of the computation region
+        map_loc = gs.read_command("g.region", flags="bg", raster=scanned_elev)
+        long = float(map_loc.split()[4:6][0].split("=")[1])
+        lat = float(map_loc.split()[4:6][1].split("=")[1])
+        # Identify the time zone of the mapped region
+        map_timezone = obj.timezone_at(lng=long, lat=lat)
+        # Convert the current day/time to the time in that timezone
+        maptime = today.astimezone(pytz.timezone(map_timezone))
+        # Extract day of year and hour
+        day = maptime.timetuple().tm_yday
+        current_time = maptime.hour
+    except Exception:
+        # Get the current day/time
+        today = datetime.datetime.today()
+        day = today.timetuple().tm_yday
+        current_time = today.hour
+        # precompute slope and aspect
+    gs.run_command(
+        "r.slope.aspect",
+        elevation=scanned_elev,
+        slope="slope",
+        aspect="aspect",
+        env=env,
+    )
+    gs.run_command(
+        "r.sun",
+        elevation=scanned_elev,
+        slope="slope",
+        aspect="aspect",
+        beam_rad="beam",
+        day=day,
+        time=current_time,
+        env=env,
+    )
+
+    # extract shade and set color to black and white
+    gs.mapcalc("shade = if(beam == 0, 0, 1)", env=env)
```

Comment:
You create shade here, but you don't display it.

Reply:
Thanks for catching that - I've updated it to just calculate and display beam.